### ISSUE TITLE
Implement abort

### ISSUE THREAD

**Issue by lukeapage** [github/fetch#547]
As per https://github.com/whatwg/fetch/pull/523

I guess it should wait till the above is merged.

**Comment by jimmywarting** [comment]
just a reference: https://github.com/mo/abortcontroller-polyfill/blob/master/src/abortcontroller.js

**Comment by coldhamix** [comment]
@lukeapage  whatwg/fetch#523 just got merged

**Comment by mislav** [comment]
I quite dig the polyfill that @jimmywarting shared. Does anyone think that this project should include that kind of polyfill? Should we try to patch native `window.fetch` if it doesn't support aborting?

**Comment by jimmywarting** [comment]
You should try calling xhr.abort to not wast bandwidth

**Comment by jmeas** [comment]
I agree with @mislav 's and @jimmywarting 's suggestions over in #572 that keeping the AbortController and AbortSignal polyfills separate from this library seems preferable.

> You should try calling xhr.abort to not wast bandwidth

It would be pretty cool to see this library update to accept a (duck-typed) `AbortSignal` and call `xhr.abort` whenever the `abort` event is dispatched.

A quick glance at the code suggests to me that this should be straightforward to implement. Perhaps the hardest part will be updating [the feature detection code](https://github.com/github/fetch/blob/7a692dc519a3a110526a26c7e252f99991510c2a/fetch.js#L4).

I wrote [a quick snippet](https://gist.github.com/jmeas/39546faa735779051bd00c3fe2a136f9) that seems like a step in the right direction. I tested it in Safari, Chrome, and Firefox, and got the expected results (no, no, yes, respectively).

What do folks think? Should I code up a PR that implements something along these lines?
